node/sc/subbridge.go

```diff
@@ -191,13 +193,14 @@ func NewSubBridge(ctx *node.ServiceContext, config *SCConfig) (*SubBridge, error
 		chainCh:        make(chan blockchain.ChainEvent, chainEventChanSize),
 		logsCh:         make(chan []*types.Log, chainLogChanSize),
 		// txCh:            make(chan blockchain.NewTxsEvent, transactionChanSize),
-		requestEventCh: make(chan *RequestValueTransferEvent, requestEventChanSize),
-		handleEventCh:  make(chan *HandleValueTransferEvent, handleEventChanSize),
-		quitSync:       make(chan struct{}),
-		maxPeers:       config.MaxPeer,
-		onAnchoringTx:  config.Anchoring,
-		bootFail:       false,
-		rpcSendCh:      make(chan []byte),
+		chanReqVTev:        make(chan *RequestValueTransferEvent, chanReqVTevanSize),
+		chanReqVTencodedEv: make(chan *RequestValueTransferEncodedEvent, chanReqVTevanSize),
```

Comment:
Since now we support 2 kinds of requestVTEvent, if you create two channels like this, the total capacity will increase from 10,000 to 20,000. Is this intended?

Reply:
Two channels have different capacities. They have rooms for size 10000 independently. Could you explain what `total capacity is? The first channel is more used than the newly added channel as it is used for ERC-20 token transfer.

Reply:
In subbridge loop, it can process by receiving events from `chanReqVTev` and `chanReqVTencodedev` respectively. I understood that up to 20,000 requestVTevents that can be processed can be stored in runtime memory.

Reply:
The value 10000 came from the origin code. I didn't increase or decrease it.

TL;DR,
The incoming events are immediately processed in the loop you've figured out in `subbridge.go`. A channel blocking delay would be created if the channel capacity is lower. By the way, the current request event handler does not handle each event with a separate go-routine. Actually, the channel capacity seems not meaningless in the current implementation. I just came up with a new improvement from your comment. We may need a benchmark test for value transfer requests per second (VTRS) in a multi-thread manner. I left details about this new improvement at #1332.